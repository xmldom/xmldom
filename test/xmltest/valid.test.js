'use strict'

const xmltest = require('xmltest')
const { getTestParser } = require('../get-test-parser')
const { generateSnapshot } = require('./generate-snapshot')

describe('xmltest/valid', () => {
	describe('standalone', () => {
		const entries = xmltest.getEntries(
			xmltest.FILTERS.VALID.SA.files,
			xmltest.FILTERS.xml
		)

		Object.entries(entries).forEach(([pathInZip, filename]) => {
			test(`should match ${filename} with snapshot`, async () => {
				const input = (await xmltest.getContent(pathInZip))
					// TODO: The DOCTYPE totally confuses xmldom :sic:
					// for now we remove it and any newlines after it so we have reasonable tests
					.replace(/^<!DOCTYPE doc \[[^\]]+]>[\r\n]*/m, '')

				const expected = await xmltest.getContent(
					xmltest.RELATED.out(pathInZip)
				)

				const { errors, parser } = getTestParser()

				const actual = parser.parseFromString(input).toString()

				expect(generateSnapshot(actual, errors, expected)).toMatchSnapshot()
			})
		})
	})
})
