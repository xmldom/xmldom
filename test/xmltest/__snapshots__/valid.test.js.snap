// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`xmltest/valid standalone should match 001.xml with snapshot 1`] = `
Object {
  "actual": "<doc/>",
  "expected": "<doc></doc>",
}
`;

exports[`xmltest/valid standalone should match 002.xml with snapshot 1`] = `
Object {
  "actual": "<doc/>",
  "expected": "<doc></doc>",
}
`;

exports[`xmltest/valid standalone should match 003.xml with snapshot 1`] = `
Object {
  "actual": "<doc/>",
  "expected": "<doc></doc>",
}
`;

exports[`xmltest/valid standalone should match 004.xml with snapshot 1`] = `
Object {
  "actual": "<doc a1=\\"v1\\"/>",
  "expected": "<doc a1=\\"v1\\"></doc>",
}
`;

exports[`xmltest/valid standalone should match 005.xml with snapshot 1`] = `
Object {
  "actual": "<doc a1=\\"v1\\"/>",
  "expected": "<doc a1=\\"v1\\"></doc>",
}
`;

exports[`xmltest/valid standalone should match 006.xml with snapshot 1`] = `
Object {
  "actual": "<doc a1=\\"v1\\"/>",
  "expected": "<doc a1=\\"v1\\"></doc>",
}
`;

exports[`xmltest/valid standalone should match 007.xml with snapshot 1`] = `
Object {
  "actual": "<doc> </doc>",
}
`;

exports[`xmltest/valid standalone should match 008.xml with snapshot 1`] = `
Object {
  "actual": "<doc>&amp;&lt;>\\"'</doc>",
  "expected": "<doc>&amp;&lt;&gt;&quot;'</doc>",
}
`;

exports[`xmltest/valid standalone should match 009.xml with snapshot 1`] = `
Object {
  "actual": "<doc> </doc>",
}
`;

exports[`xmltest/valid standalone should match 010.xml with snapshot 1`] = `
Object {
  "actual": "<doc a1=\\"v1\\"/>",
  "expected": "<doc a1=\\"v1\\"></doc>",
}
`;

exports[`xmltest/valid standalone should match 011.xml with snapshot 1`] = `
Object {
  "actual": "<doc a1=\\"v1\\" a2=\\"v2\\"/>",
  "expected": "<doc a1=\\"v1\\" a2=\\"v2\\"></doc>",
}
`;

exports[`xmltest/valid standalone should match 012.xml with snapshot 1`] = `
Object {
  "actual": "",
  "error": Array [
    "[xmldom error]	element parse error: Error: invalid attribute::
@#[line:1,col:1]",
  ],
  "expected": "<doc :=\\"v1\\"></doc>",
}
`;

exports[`xmltest/valid standalone should match 013.xml with snapshot 1`] = `
Object {
  "actual": "<doc _.-0123456789=\\"v1\\"/>",
  "expected": "<doc _.-0123456789=\\"v1\\"></doc>",
}
`;

exports[`xmltest/valid standalone should match 014.xml with snapshot 1`] = `
Object {
  "actual": "<doc abcdefghijklmnopqrstuvwxyz=\\"v1\\"/>",
  "expected": "<doc abcdefghijklmnopqrstuvwxyz=\\"v1\\"></doc>",
}
`;

exports[`xmltest/valid standalone should match 015.xml with snapshot 1`] = `
Object {
  "actual": "<doc ABCDEFGHIJKLMNOPQRSTUVWXYZ=\\"v1\\"/>",
  "expected": "<doc ABCDEFGHIJKLMNOPQRSTUVWXYZ=\\"v1\\"></doc>",
}
`;

exports[`xmltest/valid standalone should match 016.xml with snapshot 1`] = `
Object {
  "actual": "<doc><?pi ?></doc>",
}
`;

exports[`xmltest/valid standalone should match 017.xml with snapshot 1`] = `
Object {
  "actual": "<doc><?pi some data ? > <??></doc>",
}
`;

exports[`xmltest/valid standalone should match 018.xml with snapshot 1`] = `
Object {
  "actual": "<doc><![CDATA[<foo>]]></doc>",
  "expected": "<doc>&lt;foo&gt;</doc>",
}
`;

exports[`xmltest/valid standalone should match 019.xml with snapshot 1`] = `
Object {
  "actual": "<doc><![CDATA[<&]]></doc>",
  "expected": "<doc>&lt;&amp;</doc>",
}
`;

exports[`xmltest/valid standalone should match 020.xml with snapshot 1`] = `
Object {
  "actual": "<doc><![CDATA[<&]>]]]></doc>",
  "expected": "<doc>&lt;&amp;]&gt;]</doc>",
}
`;

exports[`xmltest/valid standalone should match 021.xml with snapshot 1`] = `
Object {
  "actual": "<doc><!-- a comment --></doc>",
  "expected": "<doc></doc>",
}
`;

exports[`xmltest/valid standalone should match 022.xml with snapshot 1`] = `
Object {
  "actual": "<doc><!-- a comment ->--></doc>",
  "expected": "<doc></doc>",
}
`;

exports[`xmltest/valid standalone should match 023.xml with snapshot 1`] = `
Object {
  "actual": "<doc>&amp;e;</doc>",
  "error": Array [
    "[xmldom error]	entity not found:&e;
@#[line:1,col:1]",
  ],
  "expected": "<doc></doc>",
}
`;

exports[`xmltest/valid standalone should match 024.xml with snapshot 1`] = `
Object {
  "actual": "<doc>&amp;e;</doc>",
  "error": Array [
    "[xmldom error]	entity not found:&e;
@#[line:1,col:1]",
  ],
  "expected": "<doc><foo></foo></doc>",
}
`;

exports[`xmltest/valid standalone should match 025.xml with snapshot 1`] = `
Object {
  "actual": "<doc><foo/><foo/></doc>",
  "expected": "<doc><foo></foo><foo></foo></doc>",
}
`;

exports[`xmltest/valid standalone should match 026.xml with snapshot 1`] = `
Object {
  "actual": "<doc><foo/><foo/></doc>",
  "expected": "<doc><foo></foo><foo></foo></doc>",
}
`;

exports[`xmltest/valid standalone should match 027.xml with snapshot 1`] = `
Object {
  "actual": "<doc><foo/><foo/></doc>",
  "expected": "<doc><foo></foo><foo></foo></doc>",
}
`;

exports[`xmltest/valid standalone should match 028.xml with snapshot 1`] = `
Object {
  "actual": "<?xml version=\\"1.0\\"?>
<doc/>",
  "expected": "<doc></doc>",
}
`;

exports[`xmltest/valid standalone should match 029.xml with snapshot 1`] = `
Object {
  "actual": "<?xml version='1.0'?>
<doc/>",
  "expected": "<doc></doc>",
}
`;

exports[`xmltest/valid standalone should match 030.xml with snapshot 1`] = `
Object {
  "actual": "<?xml version = \\"1.0\\"?>
<doc/>",
  "expected": "<doc></doc>",
}
`;

exports[`xmltest/valid standalone should match 031.xml with snapshot 1`] = `
Object {
  "actual": "<?xml version='1.0' encoding=\\"UTF-8\\"?>
<doc/>",
  "expected": "<doc></doc>",
}
`;

exports[`xmltest/valid standalone should match 032.xml with snapshot 1`] = `
Object {
  "actual": "<?xml version='1.0' standalone='yes'?>
<doc/>",
  "expected": "<doc></doc>",
}
`;

exports[`xmltest/valid standalone should match 033.xml with snapshot 1`] = `
Object {
  "actual": "<?xml version='1.0' encoding=\\"UTF-8\\" standalone='yes'?>
<doc/>",
  "expected": "<doc></doc>",
}
`;

exports[`xmltest/valid standalone should match 034.xml with snapshot 1`] = `
Object {
  "actual": "<doc/>",
  "expected": "<doc></doc>",
}
`;

exports[`xmltest/valid standalone should match 035.xml with snapshot 1`] = `
Object {
  "actual": "<doc/>",
  "expected": "<doc></doc>",
}
`;

exports[`xmltest/valid standalone should match 036.xml with snapshot 1`] = `
Object {
  "actual": "<doc/>
<?pi data?>",
  "expected": "<doc></doc><?pi data?>",
}
`;

exports[`xmltest/valid standalone should match 037.xml with snapshot 1`] = `
Object {
  "actual": "<doc/>
<!-- comment -->",
  "expected": "<doc></doc>",
}
`;

exports[`xmltest/valid standalone should match 038.xml with snapshot 1`] = `
Object {
  "actual": "<!-- comment -->
<doc/>",
  "expected": "<doc></doc>",
}
`;

exports[`xmltest/valid standalone should match 039.xml with snapshot 1`] = `
Object {
  "actual": "<?pi data?>
<doc/>",
  "expected": "<?pi data?><doc></doc>",
}
`;

exports[`xmltest/valid standalone should match 040.xml with snapshot 1`] = `
Object {
  "actual": "<doc a1=\\"&quot;&lt;&amp;>'\\"/>",
  "expected": "<doc a1=\\"&quot;&lt;&amp;&gt;'\\"></doc>",
}
`;

exports[`xmltest/valid standalone should match 041.xml with snapshot 1`] = `
Object {
  "actual": "<doc a1=\\"A\\"/>",
  "expected": "<doc a1=\\"A\\"></doc>",
}
`;

exports[`xmltest/valid standalone should match 042.xml with snapshot 1`] = `
Object {
  "actual": "<doc>A</doc>",
}
`;

exports[`xmltest/valid standalone should match 043.xml with snapshot 1`] = `
Object {
  "actual": "<doc a1=\\"foo
bar\\"/>",
  "expected": "<doc a1=\\"foo bar\\"></doc>",
}
`;

exports[`xmltest/valid standalone should match 044.xml with snapshot 1`] = `
Object {
  "actual": "<doc>
<e a3=\\"v3\\"/>
<e a1=\\"w1\\"/>
<e a2=\\"w2\\" a3=\\"v3\\"/>
</doc>",
  "expected": "<doc>&#10;<e a1=\\"v1\\" a2=\\"v2\\" a3=\\"v3\\"></e>&#10;<e a1=\\"w1\\" a2=\\"v2\\"></e>&#10;<e a1=\\"v1\\" a2=\\"w2\\" a3=\\"v3\\"></e>&#10;</doc>",
}
`;

exports[`xmltest/valid standalone should match 045.xml with snapshot 1`] = `
Object {
  "actual": "<doc/>",
  "expected": "<doc a1=\\"v1\\"></doc>",
}
`;

exports[`xmltest/valid standalone should match 046.xml with snapshot 1`] = `
Object {
  "actual": "<doc/>",
  "expected": "<doc a1=\\"v1\\" a2=\\"v2\\"></doc>",
}
`;

exports[`xmltest/valid standalone should match 047.xml with snapshot 1`] = `
Object {
  "actual": "<doc>X
Y</doc>",
  "expected": "<doc>X&#10;Y</doc>",
}
`;

exports[`xmltest/valid standalone should match 048.xml with snapshot 1`] = `
Object {
  "actual": "<doc>]</doc>",
}
`;

exports[`xmltest/valid standalone should match 049.xml with snapshot 1`] = `
Object {
  "actual": "{!0!}/{!0!}d{!0!}o{!0!}c{!0!}>{!0!}
{!0!}
{!0!}",
  "error": Array [
    "[xmldom error]	element parse error: Error: invalid tagName:
@#[line:1,col:3]",
    "[xmldom error]	element parse error: Error: invalid tagName:
@#[line:3,col:2]",
    "[xmldom error]	element parse error: Error: invalid tagName:
@#[line:7,col:2]",
    "[xmldom error]	element parse error: Error: invalid tagName:
@#[line:7,col:14]",
  ],
  "expected": "<doc>£</doc>",
}
`;

exports[`xmltest/valid standalone should match 050.xml with snapshot 1`] = `
Object {
  "actual": "{!0!}/{!0!}d{!0!}o{!0!}c{!0!}>{!0!}
{!0!}
{!0!}",
  "error": Array [
    "[xmldom error]	element parse error: Error: invalid tagName:
@#[line:1,col:3]",
    "[xmldom error]	element parse error: Error: invalid tagName:
@#[line:3,col:2]",
    "[xmldom error]	element parse error: Error: invalid tagName:
@#[line:7,col:2]",
    "[xmldom error]	element parse error: Error: invalid tagName:
@#[line:7,col:22]",
  ],
  "expected": "<doc>เจมส์</doc>",
}
`;

exports[`xmltest/valid standalone should match 051.xml with snapshot 1`] = `
Object {
  "actual": "{!0!}/{!0!}@!*L>{!0!}
{!0!}
{!0!}",
  "error": Array [
    "[xmldom error]	element parse error: Error: invalid tagName:
@#[line:1,col:3]",
    "[xmldom error]	element parse error: Error: invalid tagName:
@#[line:3,col:2]",
    "[xmldom error]	element parse error: Error: invalid tagName:
@#[line:7,col:2]",
    "[xmldom error]	element parse error: Error: invalid tagName:
@#[line:7,col:16]",
  ],
  "expected": "<เจมส์></เจมส์>",
}
`;

exports[`xmltest/valid standalone should match 052.xml with snapshot 1`] = `
Object {
  "actual": "<doc>𐀀􏿽</doc>",
}
`;

exports[`xmltest/valid standalone should match 053.xml with snapshot 1`] = `
Object {
  "actual": "<doc>&amp;e;</doc>",
  "error": Array [
    "[xmldom error]	entity not found:&e;
@#[line:1,col:1]",
  ],
  "expected": "<doc><e></e></doc>",
}
`;

exports[`xmltest/valid standalone should match 054.xml with snapshot 1`] = `
Object {
  "actual": "<doc/>",
  "expected": "<doc></doc>",
}
`;

exports[`xmltest/valid standalone should match 055.xml with snapshot 1`] = `
Object {
  "actual": "<?pi data?>
<doc/>",
  "expected": "<?pi data?><doc></doc>",
}
`;

exports[`xmltest/valid standalone should match 056.xml with snapshot 1`] = `
Object {
  "actual": "<doc>A</doc>",
}
`;

exports[`xmltest/valid standalone should match 057.xml with snapshot 1`] = `
Object {
  "actual": "<doc/>",
  "expected": "<doc></doc>",
}
`;

exports[`xmltest/valid standalone should match 058.xml with snapshot 1`] = `
Object {
  "actual": "<doc a1=\\" 1  	2 	\\"/>",
  "expected": "<doc a1=\\"1 2\\"></doc>",
}
`;

exports[`xmltest/valid standalone should match 059.xml with snapshot 1`] = `
Object {
  "actual": "<doc>
<e a1=\\"v1\\" a2=\\"v2\\" a3=\\"v3\\"/>
<e a1=\\"w1\\" a2=\\"v2\\"/>
<e a1=\\"v1\\" a2=\\"w2\\" a3=\\"v3\\"/>
</doc>",
  "expected": "<doc>&#10;<e a1=\\"v1\\" a2=\\"v2\\" a3=\\"v3\\"></e>&#10;<e a1=\\"w1\\" a2=\\"v2\\"></e>&#10;<e a1=\\"v1\\" a2=\\"w2\\" a3=\\"v3\\"></e>&#10;</doc>",
}
`;

exports[`xmltest/valid standalone should match 060.xml with snapshot 1`] = `
Object {
  "actual": "<doc>X
Y</doc>",
  "expected": "<doc>X&#10;Y</doc>",
}
`;

exports[`xmltest/valid standalone should match 061.xml with snapshot 1`] = `
Object {
  "actual": "<doc>£</doc>",
}
`;

exports[`xmltest/valid standalone should match 062.xml with snapshot 1`] = `
Object {
  "actual": "<doc>เจมส์</doc>",
}
`;

exports[`xmltest/valid standalone should match 063.xml with snapshot 1`] = `
Object {
  "actual": "<!DOCTYPE เจมส์><เจมส์/>",
  "expected": "<เจมส์></เจมส์>",
}
`;

exports[`xmltest/valid standalone should match 064.xml with snapshot 1`] = `
Object {
  "actual": "<doc>𐀀􏿽</doc>",
}
`;

exports[`xmltest/valid standalone should match 065.xml with snapshot 1`] = `
Object {
  "actual": "<doc/>",
  "expected": "<doc></doc>",
}
`;

exports[`xmltest/valid standalone should match 066.xml with snapshot 1`] = `
Object {
  "actual": "<doc a1=\\"&amp;e1;\\"/>",
  "error": Array [
    "[xmldom error]	entity not found:&e1;
@#[line:1,col:1]",
  ],
  "expected": "<doc a1=\\"&quot;\\"></doc>",
}
`;

exports[`xmltest/valid standalone should match 067.xml with snapshot 1`] = `
Object {
  "actual": "<doc>
</doc>",
  "expected": "<doc>&#13;</doc>",
}
`;

exports[`xmltest/valid standalone should match 068.xml with snapshot 1`] = `
Object {
  "actual": "<doc>&amp;e;</doc>",
  "error": Array [
    "[xmldom error]	entity not found:&e;
@#[line:1,col:1]",
  ],
  "expected": "<doc>&#13;</doc>",
}
`;

exports[`xmltest/valid standalone should match 069.xml with snapshot 1`] = `
Object {
  "actual": "<doc/>",
  "expected": "<doc></doc>",
}
`;

exports[`xmltest/valid standalone should match 070.xml with snapshot 1`] = `
Object {
  "actual": "<doc/>",
  "expected": "<doc></doc>",
}
`;

exports[`xmltest/valid standalone should match 071.xml with snapshot 1`] = `
Object {
  "actual": "<doc/>",
  "expected": "<doc></doc>",
}
`;

exports[`xmltest/valid standalone should match 072.xml with snapshot 1`] = `
Object {
  "actual": "<doc/>",
  "expected": "<doc></doc>",
}
`;

exports[`xmltest/valid standalone should match 073.xml with snapshot 1`] = `
Object {
  "actual": "<doc/>",
  "expected": "<doc></doc>",
}
`;

exports[`xmltest/valid standalone should match 074.xml with snapshot 1`] = `
Object {
  "actual": "<doc/>",
  "expected": "<doc></doc>",
}
`;

exports[`xmltest/valid standalone should match 075.xml with snapshot 1`] = `
Object {
  "actual": "<doc/>",
  "expected": "<doc></doc>",
}
`;

exports[`xmltest/valid standalone should match 076.xml with snapshot 1`] = `
Object {
  "actual": "<doc/>",
  "expected": "<doc></doc>",
}
`;

exports[`xmltest/valid standalone should match 077.xml with snapshot 1`] = `
Object {
  "actual": "<doc/>",
  "expected": "<doc></doc>",
}
`;

exports[`xmltest/valid standalone should match 078.xml with snapshot 1`] = `
Object {
  "actual": "<doc a=\\"v\\"/>",
  "expected": "<doc a=\\"v\\"></doc>",
}
`;

exports[`xmltest/valid standalone should match 079.xml with snapshot 1`] = `
Object {
  "actual": "<doc a=\\"v\\"/>",
  "expected": "<doc a=\\"v\\"></doc>",
}
`;

exports[`xmltest/valid standalone should match 080.xml with snapshot 1`] = `
Object {
  "actual": "<doc/>",
  "expected": "<doc a=\\"v\\"></doc>",
}
`;

exports[`xmltest/valid standalone should match 081.xml with snapshot 1`] = `
Object {
  "actual": "<doc><a/><b/><c><a/></c></doc>",
  "expected": "<doc><a></a><b></b><c><a></a></c></doc>",
}
`;

exports[`xmltest/valid standalone should match 082.xml with snapshot 1`] = `
Object {
  "actual": "<doc/>",
  "expected": "<doc></doc>",
}
`;

exports[`xmltest/valid standalone should match 083.xml with snapshot 1`] = `
Object {
  "actual": "<doc/>",
  "expected": "<doc></doc>",
}
`;

exports[`xmltest/valid standalone should match 084.xml with snapshot 1`] = `
Object {
  "actual": "<doc/>",
  "expected": "<doc></doc>",
}
`;

exports[`xmltest/valid standalone should match 085.xml with snapshot 1`] = `
Object {
  "actual": "<doc>&amp;e;</doc>",
  "error": Array [
    "[xmldom error]	entity not found:&e;
@#[line:1,col:1]",
  ],
  "expected": "<doc></doc>",
}
`;

exports[`xmltest/valid standalone should match 086.xml with snapshot 1`] = `
Object {
  "actual": "<doc>&amp;e;</doc>",
  "error": Array [
    "[xmldom error]	entity not found:&e;
@#[line:1,col:1]",
  ],
  "expected": "<doc></doc>",
}
`;

exports[`xmltest/valid standalone should match 087.xml with snapshot 1`] = `
Object {
  "actual": "<doc>&amp;e;</doc>",
  "error": Array [
    "[xmldom error]	entity not found:&e;
@#[line:1,col:1]",
  ],
  "expected": "<doc><foo></foo></doc>",
}
`;

exports[`xmltest/valid standalone should match 088.xml with snapshot 1`] = `
Object {
  "actual": "<doc>&amp;e;</doc>",
  "error": Array [
    "[xmldom error]	entity not found:&e;
@#[line:1,col:1]",
  ],
  "expected": "<doc>&lt;foo&gt;</doc>",
}
`;

exports[`xmltest/valid standalone should match 089.xml with snapshot 1`] = `
Object {
  "actual": "<doc>&amp;e;</doc>",
  "error": Array [
    "[xmldom error]	entity not found:&e;
@#[line:1,col:1]",
  ],
  "expected": "<doc>𐀀􏿽􏿿</doc>",
}
`;

exports[`xmltest/valid standalone should match 090.xml with snapshot 1`] = `
Object {
  "actual": "<doc/>",
  "expected": "<doc></doc>",
}
`;

exports[`xmltest/valid standalone should match 091.xml with snapshot 1`] = `
Object {
  "actual": "<doc/>",
  "expected": "<doc a=\\"e\\"></doc>",
}
`;

exports[`xmltest/valid standalone should match 092.xml with snapshot 1`] = `
Object {
  "actual": "<doc>
<a/>
    <a/>	<a/>


</doc>",
  "expected": "<doc>&#10;<a></a>&#10;    <a></a>&#9;<a></a>&#10;&#10;&#10;</doc>",
}
`;

exports[`xmltest/valid standalone should match 093.xml with snapshot 1`] = `
Object {
  "actual": "<doc>


</doc>",
  "expected": "<doc>&#10;&#10;&#10;</doc>",
}
`;

exports[`xmltest/valid standalone should match 094.xml with snapshot 1`] = `
Object {
  "actual": "<doc/>",
  "expected": "<doc a1=\\"%e;\\"></doc>",
}
`;

exports[`xmltest/valid standalone should match 095.xml with snapshot 1`] = `
Object {
  "actual": "<doc a1=\\"1  2\\"/>",
  "expected": "<doc a1=\\"1  2\\"></doc>",
}
`;

exports[`xmltest/valid standalone should match 096.xml with snapshot 1`] = `
Object {
  "actual": "<doc/>",
  "expected": "<doc a1=\\"1 2\\"></doc>",
}
`;

exports[`xmltest/valid standalone should match 097.xml with snapshot 1`] = `
Object {
  "actual": "<doc/>",
  "expected": "<doc a1=\\"v1\\"></doc>",
}
`;

exports[`xmltest/valid standalone should match 098.xml with snapshot 1`] = `
Object {
  "actual": "<doc><?pi x
y?></doc>",
  "expected": "<doc><?pi x
y?></doc>",
}
`;

exports[`xmltest/valid standalone should match 099.xml with snapshot 1`] = `
Object {
  "actual": "<?xml version=\\"1.0\\" encoding=\\"utf-8\\"?>
<doc/>",
  "expected": "<doc></doc>",
}
`;

exports[`xmltest/valid standalone should match 100.xml with snapshot 1`] = `
Object {
  "actual": "<doc/>",
  "expected": "<doc></doc>",
}
`;

exports[`xmltest/valid standalone should match 101.xml with snapshot 1`] = `
Object {
  "actual": "<doc/>",
  "expected": "<doc></doc>",
}
`;

exports[`xmltest/valid standalone should match 102.xml with snapshot 1`] = `
Object {
  "actual": "<doc a=\\"&quot;\\"/>",
  "expected": "<doc a=\\"&quot;\\"></doc>",
}
`;

exports[`xmltest/valid standalone should match 103.xml with snapshot 1`] = `
Object {
  "actual": "<doc>&lt;doc></doc>",
  "expected": "<doc>&lt;doc&gt;</doc>",
}
`;

exports[`xmltest/valid standalone should match 104.xml with snapshot 1`] = `
Object {
  "actual": "<doc a=\\"x	y\\"/>",
  "expected": "<doc a=\\"x y\\"></doc>",
}
`;

exports[`xmltest/valid standalone should match 105.xml with snapshot 1`] = `
Object {
  "actual": "<doc a=\\"x	y\\"/>",
  "expected": "<doc a=\\"x&#9;y\\"></doc>",
}
`;

exports[`xmltest/valid standalone should match 106.xml with snapshot 1`] = `
Object {
  "actual": "<doc a=\\"x
y\\"/>",
  "expected": "<doc a=\\"x&#10;y\\"></doc>",
}
`;

exports[`xmltest/valid standalone should match 107.xml with snapshot 1`] = `
Object {
  "actual": "<doc a=\\"x
y\\"/>",
  "expected": "<doc a=\\"x&#13;y\\"></doc>",
}
`;

exports[`xmltest/valid standalone should match 108.xml with snapshot 1`] = `
Object {
  "actual": "<doc a=\\"x&amp;e;y\\"/>",
  "error": Array [
    "[xmldom error]	entity not found:&e;
@#[line:1,col:1]",
  ],
  "expected": "<doc a=\\"x y\\"></doc>",
}
`;

exports[`xmltest/valid standalone should match 109.xml with snapshot 1`] = `
Object {
  "actual": "<doc a=\\"\\"/>",
  "expected": "<doc a=\\"\\"></doc>",
}
`;

exports[`xmltest/valid standalone should match 110.xml with snapshot 1`] = `
Object {
  "actual": "<doc a=\\"x&amp;e;y\\"/>",
  "error": Array [
    "[xmldom error]	entity not found:&e;
@#[line:1,col:1]",
  ],
  "expected": "<doc a=\\"x  y\\"></doc>",
}
`;

exports[`xmltest/valid standalone should match 111.xml with snapshot 1`] = `
Object {
  "actual": "<doc a=\\" x  y \\"/>",
  "expected": "<doc a=\\"x y\\"></doc>",
}
`;

exports[`xmltest/valid standalone should match 112.xml with snapshot 1`] = `
Object {
  "actual": "<doc><a/></doc>",
  "expected": "<doc><a></a></doc>",
}
`;

exports[`xmltest/valid standalone should match 113.xml with snapshot 1`] = `
Object {
  "actual": "<doc/>",
  "expected": "<doc></doc>",
}
`;

exports[`xmltest/valid standalone should match 114.xml with snapshot 1`] = `
Object {
  "actual": "<!DOCTYPE doc><doc>&amp;e;</doc>",
  "error": Array [
    "[xmldom error]	entity not found:&e;
@#[line:5,col:1]",
  ],
  "expected": "<doc>&amp;foo;</doc>",
}
`;

exports[`xmltest/valid standalone should match 115.xml with snapshot 1`] = `
Object {
  "actual": "<doc>&amp;e1;</doc>",
  "error": Array [
    "[xmldom error]	entity not found:&e1;
@#[line:1,col:1]",
  ],
  "expected": "<doc>v</doc>",
}
`;

exports[`xmltest/valid standalone should match 116.xml with snapshot 1`] = `
Object {
  "actual": "<doc><![CDATA[
]]></doc>",
  "expected": "<doc>&#10;</doc>",
}
`;

exports[`xmltest/valid standalone should match 117.xml with snapshot 1`] = `
Object {
  "actual": "<!DOCTYPE doc><doc>&amp;rsqb;</doc>",
  "error": Array [
    "[xmldom error]	entity not found:&rsqb;
@#[line:5,col:1]",
  ],
  "expected": "<doc>]</doc>",
}
`;

exports[`xmltest/valid standalone should match 118.xml with snapshot 1`] = `
Object {
  "actual": "<!DOCTYPE doc><doc>&amp;rsqb;</doc>",
  "error": Array [
    "[xmldom error]	entity not found:&rsqb;
@#[line:5,col:1]",
  ],
  "expected": "<doc>]]</doc>",
}
`;

exports[`xmltest/valid standalone should match 119.xml with snapshot 1`] = `
Object {
  "actual": "<doc><!-- -á --></doc>",
  "expected": "<doc></doc>",
}
`;
